### PR TITLE
C99 compatibility fixes

### PR DESCRIPTION
**Pull request by fweimer-rh**
Related to:

* https://fedoraproject.org/wiki/Changes/PortingToModernC
* https://fedoraproject.org/wiki/Toolchain/PortingToModernC

### CODE REVIEW

**Review by xiaoyeli — COMMENTED** [review]
These were already added, see this commit:
https://github.com/xiaoyeli/superlu/commit/a49ee5f535c932e6fa604eb12fac3e7796231713

**Review by xiaoyeli — APPROVED** [review]

**Review by xiaoyeli — APPROVED** [review]

**Review by xiaoyeli — APPROVED** [review]

File: EXAMPLE/superlu.c [hunk 1]
```diff
@@ -20,7 +20,7 @@ at the top-level directory.
  */
 #include "slu_ddefs.h"
 
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
```
Comment (gruenich) [inline EXAMPLE/superlu.c:23]: Declaring `main` to return an `int` and not returning anything is weird. I know, it is allowed by C99, but I'd prefer adding `return EXIT_SUCCESS;` as the last line of `main` to stay consistent.

**Comment by gruenich** [comment]
I propose to drop the last commit 0970dc2 and add the return statement to 4dc22d8, then they are good to go.

**Comment by xiaoyeli** [comment]
superseded by #82

**Pull request closed without merging**